enterprise/cmd/frontend/internal/auth/githubappauth/middleware.go

```diff
@@ -187,7 +200,7 @@ func newServeMux(db edb.EnterpriseDB, prefix string, cache *rcache.Cache) http.H
 		state := query.Get("state")
 		instID := query.Get("installation_id")
 		if state == "" || instID == "" {
-			http.Error(w, "Bad request, installation_id and state query params must be present", http.StatusBadRequest)
+			http.Redirect(w, req, "/site-admin/github-apps", http.StatusFound)
```

Comment:
Can you add a comment here on why we do a redirect here? Something like:
```suggestion
			// redirecting to github apps list, since we cannot map to the app id
			http.Redirect(w, req, "/site-admin/github-apps", http.StatusFound)
```